src/asm/section.cpp

```diff
@@ -937,14 +930,14 @@ void sect_PushSection(void)
 		.scope = sym_GetCurrentSymbolScope(),
 		.offset = curOffset,
 		.loadOffset = loadOffset,
-		.unionStack = currentUnionStack,
+		.unionStack = {},
 	});
 
 	// Reset the section scope
 	currentSection = NULL;
 	currentLoadSection = NULL;
 	sym_SetCurrentSymbolScope(NULL);
-	currentUnionStack = NULL;
+	std::swap(currentUnionStack, sectionStack.front().unionStack);
```

Comment:
Consider using `std::deque::emplace_front`, which returns a reference to the inserted element ;)

Reply:
IIRC I've been using `.push_front({ .a = a, .b = b, ... })` instead of `.emplace_front(a, b, ...)` because I got an error the first time I tried that, and didn't want to bother implementing constructors for our structs. (I do use `.emplace_front()` and get the returned reference when it's default-constructed.)

Reply:
Have you tried passing a brace-init-list to `emplace`?

Reply:
Brace-init-lists don't work in `emplace` because the init list can't deduce the type it's initializing (and even if it could, the result would be the same as in `push` where it constructs an object and then moves it into the container).

The `.emplace_front(a, b, ...)` approach *would* work in C++20 because they made aggregates constructible that way, but pre-C++20 you would have to explicitly define a constructor.

Reply:
(Defining constructors in general could also be a separate PR.)

Reply:
Added to #1315.